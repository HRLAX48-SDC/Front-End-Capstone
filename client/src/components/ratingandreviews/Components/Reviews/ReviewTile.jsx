import React, { useState, useEffect} from 'react';
import moment from 'moment';
import ReviewBody from './ReviewBody.jsx';
import { Card, Stack } from 'react-bootstrap';
import API_KEY from '../../../../config/config.js';
import axios from 'axios';


function ReviewTile (props) {

  const [reviewHelpfulNum, setReviewHelpfulNum] = useState(() => props.review.helpfulness);
  const [helpfulSelection, setHelpfulSelection] = useState(false);
 
  
  const selectHelpful = () =>{
    if (!helpfulSelection) {
    axios.put(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/reviews/${props.review.review_id}/helpful`, 
    {helpfulness: reviewHelpfulNum + 1},
    { headers: { 'Authorization': `${API_KEY}` } })
    .then(res => {
      setReviewHelpfulNum(reviewHelpfulNum + 1);
      setHelpfulSelection(true);
    })
    .catch(err => console.log(err));
    }
  }

  const selectUnHelpful = () =>{
    if (!helpfulSelection) {
    axios.put(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/reviews/${props.review.review_id}/helpful`, 
    {helpfulness: reviewHelpfulNum - 1},
    { headers: { 'Authorization': `${API_KEY}` } })
    .then(res => {
      setReviewHelpfulNum(reviewHelpfulNum - 1);
      setHelpfulSelection(true);
    })
    .catch(err => console.log(err));
    }
  }

  //recommended
  //if the review is recommended display "this review is recommended"

  //make put request to report reviews...


  return (
    <>
      <Card>
        <Card.Header>
          <Stack direction="horizontal" gap={3}>
            <div>*star rating*</div>
            <div className="ms-auto">{props.review.reviewer_name}</div>
            <div>{moment(props.review.date).format('MMMM Do, YYYY')}</div>
          </Stack>
        </Card.Header>
        <Card.Body>
          <Card.Title>{props.review.title}</Card.Title>
          <Card.Text>
          {props.review.summary}
          </Card.Text>
          <ReviewBody review={props.review}/>
          <Stack direction="horizontal" gap={3}>
            <div>Helpful?</div>
            <div onClick={() => selectHelpful()}>Yes</div>
            <div onClick={() => selectUnHelpful()}>No</div>
            <div>({reviewHelpfulNum})</div>
            <div>*report*</div>
           </Stack>
        </Card.Body>
        
      </Card>
    </>
  )

}
export default ReviewTile;

