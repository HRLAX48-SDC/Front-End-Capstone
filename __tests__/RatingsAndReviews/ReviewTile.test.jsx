import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ReviewTile from '../../client/src/components/ratingandreviews/Components/Reviews/ReviewTile.jsx'

Enzyme.configure({adapter: new Adapter()});

describe('<ReviewTile />', () => {
    const review = {
        body: "Wow this product is amazing, 100/10 recommend. I own 1000 of these muahahaha. ",
        date: "2021-11-10T00:00:00.000Z",
        helpfulness: 6,
        photos: [],
        rating: 5,
        recommend: true,
        response: null,
        review_id: 1094674,
        reviewer_name: "BananaBoy123",
        summary: "Is this really the best purchase ever?"
    } 
    
    it('Should correctly render a review tile', () => {
        const reviewTile = renderer
        .create(<ReviewTile review={review}/>)
        .toJSON();
        expect(reviewTile).toMatchSnapshot()
    });

    const wrapper = mount(<ReviewTile key={review.review_id} review={review}/>);
    it('should render a way to mark a review helpful', () => {
        wrapper.find('.review-helpful').simulate('click')
        expect(review.helpfulness).toEqual(review.helpfulness)
    })
    it('should render a way to report a review', () => {
        wrapper.find('.review-report').simulate('click')
        expect(review).toEqual(review)
    })
    
   
});

