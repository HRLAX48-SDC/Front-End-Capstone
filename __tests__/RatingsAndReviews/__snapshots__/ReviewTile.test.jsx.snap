// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewTile /> Should correctly render a review tile 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <div
      className="hstack gap-3"
    >
      <span
        style={
          Object {
            "direction": "ltr",
            "display": "inline-block",
            "touchAction": "none",
          }
        }
      >
        <span
          aria-hidden="true"
          className="react-simple-star-rating"
          style={
            Object {
              "cursor": "",
              "display": "inline-block",
              "overflow": "hidden",
              "position": "relative",
              "userSelect": "none",
              "verticalAlign": "middle",
              "whiteSpace": "nowrap",
            }
          }
        >
          <span
            className="empty-icons"
            style={
              Object {
                "color": "#cccccc",
                "display": "inline-block",
              }
            }
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
          </span>
          <span
            className="filled-icons"
            style={
              Object {
                "color": "#ffbc0b",
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "transition": "",
                "whiteSpace": "nowrap",
                "width": "100%",
              }
            }
            title="100 out of 5"
          >
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
            <svg
              className="star-svg"
              fill="currentColor"
              height={25}
              style={Object {}}
              viewBox="0 0 24 24"
              width={25}
            >
              <path
                d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
                fill="currentColor"
                stroke="none"
                strokeMiterlimit="10"
                strokeWidth={0}
              />
            </svg>
          </span>
        </span>
      </span>
      <div
        className="ms-auto"
      >
        BananaBoy123
      </div>
      <div
        className="review-tile-date"
      >
        November 9th, 2021
      </div>
    </div>
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
      Is this really the best purchase ever?
    </p>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          Wow this product is amazing, 100/10 recommend. I own 1000 of these muahahaha. 
        </p>
        <div
          className="row"
        />
      </div>
    </div>
    <div
      className="hstack gap-2"
    >
      <div>
        Was this review helpful?
      </div>
      <div
        className="review-helpful"
        onClick={[Function]}
      >
        Yes
      </div>
      <div>
        (
        6
        )
      </div>
      <div
        className="review-report"
        onClick={[Function]}
      >
        Report
      </div>
      <div
        className="ms-auto"
      >
        I recommend this product. âœ”
      </div>
    </div>
  </div>
</div>
`;
